Phase4/Project/Ranker.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Project
+{
+    public class Ranker
+    {
+        private readonly Student[] _students;
+        private readonly StudentScore[] _scores;
+
+        public Ranker(Student[] students, StudentScore[] scores)
+        {
+            _students = students;
+            _scores = scores;
+        }
+
+        private List<Student> GetRankedStudents()
+        {
+            var rankedStudent = _students.OrderByDescending(student => GetStudentScore(student)).ToList();
+            return rankedStudent;
```

Comment:
_rankedStudent_ is actually redundant and not used for any computation. Therefore its a better idea to remove that and just return the right hand side of the assignment.